includes/course-theme/class-sensei-course-theme-templates.php

```diff
@@ -236,6 +380,12 @@ public function get_block_templates() {
 		$templates    = [];
 
 		foreach ( $this->file_templates as $name => $template ) {
+			// Prefill the template contents from their content files.
+			if ( isset( $template['content'] ) && file_exists( $template['content'] ) ) {
+				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- Local file usage.
+				$template['content'] = file_get_contents( $template['content'] );
+			}
+
 			$db_template     = $db_templates[ $name ] ?? null;
```

Comment:
Note that if the user customized a template (eg Modern) and then chooses a different one (Video), it will still use their custom version (of Modern) both in the editor & frontend. 

I think we'll need to do add the template name to the slug prefix (`sensei-course-theme//lesson`) or otherwise filter the search in `get_custom_templates`. 

Fine to figure it out in a different PR (we have a card, [#5538](https://github.com/Automattic/sensei/issues/5538)) since it could get complicated :)

Reply:
Noted!

I checked, and it looks like the template actually changes. Both in the editor and frontend. What doesn't change is the name in the list of templates in the editor.